kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/format/KafkaEntryFormatter.java

```diff
@@ -71,15 +73,25 @@ public MemoryRecords decode(List<Entry> entries, byte magic) {
         return builder.build();
     }
 
-    private static PulsarApi.MessageMetadata getMessageMetadataWithNumberMessages(int numMessages) {
+    private static PulsarApi.MessageMetadata getMessageMetadataWithNumberMessages(int numMessages,
+                                                                                  MemoryRecords records) {
         final PulsarApi.MessageMetadata.Builder builder = PulsarApi.MessageMetadata.newBuilder();
         builder.addProperties(PulsarApi.KeyValue.newBuilder()
                 .setKey("entry.format")
                 .setValue(EntryFormatterFactory.EntryFormat.KAFKA.name().toLowerCase())
                 .build());
         builder.setProducerName("");
         builder.setSequenceId(0L);
-        builder.setPublishTime(0L);
+        boolean res = StreamSupport.stream(records.records().spliterator(), true).anyMatch(record -> {
+            if (record.timestamp() > 0) {
+                builder.setPublishTime(record.timestamp());
+                return true;
+            }
+            return false;
+        });
```

Comment:
Even if it's a parallel stream, the `builder.setPublishTime` will be invoked for each record that has timestamp. If a `MemoryRecords` has 10000 records, it will be set for 10000 times but only one random (because it's parallel stream) timestamp will be set.

Why not just set it to `System.currentTimeMillis()`? The publish time should be associated to the whole records, not any single record.

Reply:
`System.currentTimeMillis()` make sense